components/sup/src/ctl_gateway/mod.rs

```diff
@@ -163,56 +160,65 @@ impl Write for CtlRequest {
     fn flush(&mut self) -> io::Result<()> { Ok(()) }
 
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let line = String::from_utf8(buf.to_vec()).expect("CtlRequest buffer valid utf8");
+
         // The protocol reply is destined for the client, so (for now,
         // at least), we'll apply colored output.
         //
         // `line` will also have a newline character at the end, FYI.
         let mut msg = protocol::ctl::ConsoleLine::default();
-        msg.line = String::from_utf8_lossy(buf).to_string();
-        match self.current_color_spec {
-            Some(ref spec) => {
-                msg.color = color_to_string(spec.fg());
-                msg.bold = spec.bold();
-            }
-            None => {
-                msg.color = None;
-                msg.bold = false;
-            }
-        }
+        msg.line = line.to_string();
+        msg.color = color_to_string(self.current_color_spec.fg());
+        msg.bold = self.current_color_spec.bold();
         self.reply_partial(msg);
 
         // Down here, however, we're doing double-duty by *also*
-        // sending the output to the Supervisor's output stream. In
-        // _this_ case, we want to honor whatever global logging flags
-        // have been set; in particular, if we're emitting
-        // JSON-formatted logs or if we've been instructed to not
-        // output ANSI color codes, we need to strip those codes out.
```

Comment:
Does this still work?

Reply:
Actually, it looks like this breaks. You can see this by running `hab sup run --json-logging` with current stable Habitat vs. this PR and then doing something like:

```hab svc load <some package that isn't currently installed>```

In the supervisor terminal you should see only JSON, and no color, while in the `hab svc load` terminal, you should see "normal" colored output.

With this PR, you still get colored output in the server output.

Reply:
that is definitely not good. I'll have a look

Reply:
Simple logic bug that I fixed.

Reply:
Confirmed: it works now 👍